proparse/src/main/java/org/prorefactor/proparse/antlr4/PreproEval.java

```diff
@@ -0,0 +1,612 @@
+import org.prorefactor.refactor.settings.IProparseSettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PreproEval extends PreprocessorParserBaseVisitor<Object> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PreproEval.class);
+
+  /**
+   * Main entry point returning the evaluation of the preprocessor expression
+   * 
+   * @return A Boolean object
+   */
+  @Override
+  public Boolean visitPreproIfEval(PreproIfEvalContext ctx) {
+    LOGGER.trace("Entering visitPreproIfEval()");
+    Object o = visit(ctx.expr());
+    LOGGER.trace("Exiting visitPreproIfEval() with return value '{}'", o);
+    return (o != null) && getBool(o);
+  }
+
+  // ****
+  // Expr
+  // ****
+  @Override
+  public Object visitOr(OrContext ctx) {
+    boolean b1 = getBool(visit(ctx.expr(0)));
+    boolean b2 = getBool(visit(ctx.expr(1)));
+
+    return new Boolean(b1 || b2);
+  }
+
+  @Override
+  public Object visitAnd(AndContext ctx) {
+    return new Boolean(getBool(visit(ctx.expr(0))) && getBool(visit(ctx.expr(1))));
+  }
+
+  @Override
+  public Object visitMatches(MatchesContext ctx) {
+    return matches(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitBegins(BeginsContext ctx) {
+    return new Boolean(getString(visit(ctx.expr(0))).toLowerCase().startsWith(
+        getString(visit(ctx.expr(1))).toLowerCase()));
+  }
+
+  @Override
+  public Object visitEquals(EqualsContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.EQ);
+  }
+
+  @Override
+  public Object visitNotEquals(NotEqualsContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.NE);
+  }
+
+  @Override
+  public Object visitGreaterThan(GreaterThanContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.GT);
+  }
+
+  @Override
+  public Object visitGreaterEquals(GreaterEqualsContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.GE);
+  }
+
+  @Override
+  public Object visitLesserThan(LesserThanContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.LT);
+  }
+
+  @Override
+  public Object visitLesserEquals(LesserEqualsContext ctx) {
+    return compare(visit(ctx.expr(0)), visit(ctx.expr(1)), Compare.LE);
+  }
+
+  @Override
+  public Object visitPlus(PlusContext ctx) {
+    return opPlus(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitMinus(MinusContext ctx) {
+    return opMinus(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitDivide(DivideContext ctx) {
+    return opDivide(visit(ctx.expr(0)), visit(ctx.expr(1)));
+  }
+
+  @Override
+  public Object visitModulo(ModuloContext ctx) {
+    Double m1 = getFloat(visit(ctx.expr(0))) + .5;
+    Double m2 = getFloat(visit(ctx.expr(1))) + .5;
+    return new Integer(m1.intValue() % m2.intValue());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this "Integer" constructor [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2129)